### PR TITLE
[SYCL] Enable SPV_INTEL_token_type extension

### PR DESCRIPTION
**Pull request by MrSidims**
Don't enable it for FPGA H/W yet.

Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 1]
```diff
@@ -8656,6 +8656,9 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
       // lowered to CrossWorkgroup storage class that is mapped to just
       // global address space.
       ExtArg += ",+SPV_INTEL_usm_storage_classes";
+    else
+      // Don't enable several freshly added extensions on FPGA H/W
+      ExtArg += ",+SPV_INTEL_token_type";
```
Comment (AGindinson) [inline clang/lib/Driver/ToolChains/Clang.cpp:8661]: Unless I'm missing some other condition, negation of `!isFPGAEmulationMode()` (line ) is only going to enable the extension for FPGA Emulation and bypass all other targets. The LITs are positive, though, so how exactly does this work out?

Comment (MrSidims) [reply clang/lib/Driver/ToolChains/Clang.cpp:8661]: AFAIK `FPGAEmulationMode = true` by default and only FPGA H/W setting is changing it to `false`. I believe it's better at some point reverse `FPGAEmulationMode = true` to `FPGAHardwareMode = false` , so the code like this wouldn't confuse anybody.

Comment (AGindinson) [reply clang/lib/Driver/ToolChains/Clang.cpp:8661]: Thanks, fair point. Maybe a `enum` of some sort would also be an option (to dodge any bugs in FPGA-irrelevant driver code). Could you please file a GH issue for that?

**Review by AGindinson — APPROVED** [review]

File: clang/lib/Driver/ToolChains/Clang.cpp (hunk 1 above)
Comment (MrSidims) [reply clang/lib/Driver/ToolChains/Clang.cpp:8661]: Done in https://github.com/intel/llvm/issues/4061

**Comment by MrSidims** [comment]
@bader previously the translator would assert out in case of a token type appearance in LLVM IR and since we haven't seen any asserts I don't expect to see any regressions.

**Pull request merged**